src/pip/_internal/utils/compatibility_tags.py

```diff
@@ -130,13 +130,10 @@ def get_supported(
 
     interpreter = _get_custom_interpreter(impl, version)
 
-    abis = None  # type: Optional[List[str]]
-    if abi is not None:
-        abis = [abi]
-
-    platforms = None  # type: Optional[List[str]]
-    if platform is not None:
-        platforms = _get_custom_platforms(platform)
+    if platforms and len(platforms) == 1:
+        # Only expand list of platforms if a single platform was provided.
+        # Otherwise, assume that the list provided is comprehensive.
+        platforms = _get_custom_platforms(platforms[0])
```

Comment:
Why not call `_get_custom_platforms` on all provided `platforms` ?

I'd find it strange that `pip download --platform manylinux2010 foobar` would successfully find a `manylinux1_x86_64` whel but ` `pip download --platform manylinux2010 --platform manylinux2014 foobar` wouldn't.

Reply:
@xavfernandez I think this is a bit tricky.

I think it's reasonable to assume that anybody providing multiple `--platform` invocations is a fairly expert user, who is targeting a particular python interpreter different than the one through which `pip` is running. Therefore, the user should be the authority on what is and is not supported.

There may be edge-cases in which `manylinux2014` wheels are acceptable, but `manylinux1` wheels are not. For instance, the `manylinux2014` and `manylinux2010` specifications don't allow for binaries to link against the `ncurses` library, whereas `manylinux1` does allow this library. If a user knows that their target environment does not provide `ncurses`, then they might write

```
pip download \
    --platform manylinux2014  --platform manylinux2010 --platform any \
    --abi cp39m --abi abi3 --abi none \
    --python-version 39 \
    --no-deps \
    hiredis==0.3.1
```

To avoid downloading wheels built for `manylinux1`.

This won't be an issue for most users, but it could come up in more complex deployments (e.g. large monorepos), which (I think) is the setting in which this feature is most likely to be useful.

What do you think? We could always add platform-expansion here later - It's probably more difficult to take it back, afterwards.

Reply:
Another question is one of precedence. Given the following two commands:

```
1. pip download --platform manylinux2014_x86_64 --platform linux_x86_64 foobar==0.1.0

2. pip download --platform manylinux_2014_x86_64 --platform any foobar==0.1.0
```

where the available wheels are built for `[linux_x86_64, manylinux2010_x86_64, any]`, what should each command download?

If we perform `manylinux`-expansion, I would expect for `1` to download `linux_86_64` (since I explicitly declared that platform to be acceptable), whereas I would expect for `2` to download `manylinux2010_x86_64` (since `any` should probably be lowest precedence).

On the other hand, one might argue that `manylinux2014_x86_64` should expand "in-place" in the precedence-list, to include all three `manylinux`-platforms. In that case, `1` should download `manylinux2010_x86_64`.

My intention isn't to argue which approach is more "correct" - Instead, it's to point out that the outcome is non-obvious. If we instead tell `pip` to just trust what the user provided, then this question melts away.

Reply:
I would expects the `--platform` values be simply union-ed, and all expansions done in-place. The wheels downloaded would typically be installed by pip install on a specific platform, which would again filter the wheels. Incompatible wheels downloaded previously wouldn’t really make any practical differences (except some additional local disk space).

Reply:
Fair enough, I suppose I don't feel strongly in either direction.

@uranusjr @xavfernandez I've pushed a change to implement this - Let me know what you think.